File: fizzbuzz_spec.py
import unittest
from should_dsl import should
from fizzbuzz import fizzbuzz

class TestFizzBuzz(unittest.TestCase):
    
    def test_tres_deve_retornar_fizz(self):
        fizzbuzz(3) |should| equal_to("fizz")
        
    def test_cinco_deve_retornar_buzz(self):
        fizzbuzz(5) |should| equal_to("buzz")
        
    def test_um_deve_retornar_um(self):
        fizzbuzz(1) |should| equal_to(1)
        
    def test_seis_deve_retornar_fizz(self):
        fizzbuzz(6) |should| equal_to("fizz")
       
    def test_quatro_deve_retornar_quatro(self):
        fizzbuzz(4) |should| equal_to(4)
       
    def test_quinze_deve_retornar_fizzbuzz(self):
        fizzbuzz(15) |should| equal_to("fizzbuzz")
     
                 

File: 2012.03.23.fizzbuzz.ruby/rspec/fizzbuzz_spec.rb
require "../fizzbuzz.rb"

describe "Teste do FizzBuzz" do

	it "com o numero 3 deve retornar fizz" do
		fizzbuzz(3).should == "fizz"
  end

  it "com o numero 5 deve retornar buzz" do
    fizzbuzz(5).should == "buzz"
	end

	it "com o numero 15 deve retornar fizzbuzz" do
		fizzbuzz(15).should == "fizzbuzz"
	end

	it "com o numero 30 deve retornar fizzbuzz" do
		fizzbuzz(30).should == "fizzbuzz"	
	end

	it "com o numero 45 deve retornar fizzbuzz" do
	   fizzbuzz(45).should == "fizzbuzz"
	end

	it "com o numero 37 não deve retornar nada" do
		fizzbuzz(37).should == ""
  end

	it "com o numero 60 deve retornar fizzbuzz" do
		fizzbuzz(60).should == "fizzbuzz"
	end

	it "com o numero 75 deve retornar fizzbuzz" do
		fizzbuzz(75).should == "fizzbuzz"
	end
end
